app/src/main/java/io/left/meshim/activities/ChatActivity.java

```diff
@@ -77,6 +104,47 @@ protected void onCreate(Bundle savedInstanceState) {
 
     }
 
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        switch (requestCode) {
+            case Constant.REQUEST_CODE_PICK_FILE:
+                if (resultCode == RESULT_OK) {
+                    ArrayList<NormalFile> list = data.getParcelableArrayListExtra(Constant.RESULT_PICK_FILE);
+                    // always get the first file
+                    NormalFile file = list.get(0);
```

Comment:
Can this ever be an empty list?

Reply:
this only get executed when `(resultCode == RESULT_OK)` meaning user selected a file. otherwise it will not execute this code.